extensions/metadata/interface/src/main/java/org/exist/storage/md/RestoreFromBackup.java

```diff
@@ -0,0 +1,158 @@
+/*
+ *  eXist Open Source Native XML Database
+ *  Copyright (C) 2001-2016 The eXist Project
+ *  http://exist-db.org
+ *
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public License
+ *  as published by the Free Software Foundation; either version 2
+ *  of the License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.exist.storage.md;
+
+import org.exist.backup.BackupDescriptor;
+import org.exist.backup.SystemImport;
+import org.exist.dom.QName;
+import org.exist.util.EXistInputSource;
+import org.exist.xquery.*;
+import org.exist.xquery.value.*;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.XMLReader;
+import org.xml.sax.helpers.DefaultHandler;
+
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Stack;
+
+import static org.exist.storage.md.MetaData.NAMESPACE_URI;
+import static org.exist.storage.md.MetaData.PREFIX;
+
+import static org.exist.storage.md.MDStorageManager.*;
+
+/**
+ * @author <a href="mailto:shabanovd@gmail.com">Dmitriy Shabanov</a>
+ */
+public class RestoreFromBackup extends BasicFunction {
```

Comment:
at that PR eXist's standard backup will have metadata storage data too and sometime it's useful to restore metadata storage data only (much faster)